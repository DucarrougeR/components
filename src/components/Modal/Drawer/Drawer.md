Drawers are modal overlays that provide contextual information. They are frequantly used for creation or management action by the user. The Drawer blocks interactions with the other content until dismissed either through clicking or tapping on the Backdrop or by selecting a close or completion action within the Drawer.

```js
<Drawer content="This is some content in the Drawer">
  <Button>Open Drawer</Button>
</Drawer>
```

---

### Composed Drawer

Drawers can be constructed with Modal helpers to create a complex layout a fixed-position header and footer as well as a content area that indicates overflow and allows the user to scroll through its contents.

```js
const ModalContext = require('../ModalContext').ModalContext

const boxProps = {
  alignItems: 'center',
  p: 'xxlarge',
  mb: 'xlarge',
  display: 'flex',
  height: '40%',
  justifyContent: 'center',
  bg: 'lavender',
}

const content = (
  <>
    <ModalHeader>
      <Heading fontWeight="semiBold">Some Things to Look at</Heading>
    </ModalHeader>
    <ModalContent>
      <Box {...boxProps}>Thing One</Box>
      <Box {...boxProps}>Thing Two</Box>
      <Box {...boxProps}>Thing Three</Box>
    </ModalContent>
    <ModalContext.Consumer>
      {({ close }) => (
        <ModalFooter>
          <Button onClick={close} variant="transparent" mr="small">
            Cancel
          </Button>
          <Button
            onClick={() => {
              alert("Oh yeah! You're done")
              close && close()
            }}
          >
            All Done
          </Button>
        </ModalFooter>
      )}
    </ModalContext.Consumer>
  </>
)
;<Drawer content={content}>
  <Button variant="outline">See some things, in a Drawer!</Button>
</Drawer>
```
