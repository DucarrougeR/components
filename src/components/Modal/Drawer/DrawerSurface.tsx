import React, { FunctionComponent } from 'react'
import styled, { StyledComponent } from 'styled-components'
import { ModalSurface, ModalSurfaceProps } from '../ModalSurface'

export type DrawerSurfaceComponentType = FunctionComponent<ModalSurfaceProps>
export type StyledDrawerSurfaceComponentType = StyledComponent<
  DrawerSurfaceComponentType,
  ModalSurfaceProps
>

export const DrawerSurface: DrawerSurfaceComponentType = ({
  children,
  width = '30rem',
  anchor = 'right',
  ...props
}) => {
  return (
    <Surface
      height="100%"
      maxWidth="100%"
      width={width}
      anchor={anchor}
      {...props}
    >
      {children}
    </Surface>
  )
}

// Shadow here is designed to match theme.shadows[3] but with a single left-side shadow

const Surface: StyledDrawerSurfaceComponentType = styled<
  DrawerSurfaceComponentType
>(ModalSurface)`
  box-shadow: -18px 0 18px -18px rgba(0, 0, 0, 0.12);

  &.entering,
  &.exiting {
    opacity: 0.01;
    transform: translateX(100%);
  }

  &.exited {
    opacity: 1;
    transform: translateX(0%);
  }
`
