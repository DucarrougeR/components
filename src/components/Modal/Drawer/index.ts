export * from './Drawer'
export * from './DrawerSurface'
