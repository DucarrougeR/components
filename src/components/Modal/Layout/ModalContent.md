Using the `ModalContent` component quickly gives your content consistent spacing inside a modal component as well as managing overflow of content of the dialog is taller than the viewport.

ModalContent supports all [`Box`](/#!/Box) properties.

```js
import { Box } from '../../Box'
import { Paragraph } from '../../Text'
import { ModalContent } from '../Layout'

;<Box height="10rem" display="flex" bg="white" p="large">
  <ModalContent>
    <Box height="2rem" bg="rebeccapurple" />
    <Paragraph>Scroll down here...</Paragraph>
  </ModalContent>
</Box>
```

#### Content Overflows

If content overflows the component's height an visual indicator of content that extends beyond the visible edges is added (border on top and shadow on the bottom).

```js
import { Box } from '../../Box'
import { Paragraph } from '../../Text'
import { ModalContent } from '../Layout'

;<Box height="10rem" display="flex" bg="white" p="large">
  <ModalContent>
    <Box height="6rem" bg="rebeccapurple" />
    <Paragraph>Scroll down here...</Paragraph>
  </ModalContent>
</Box>
```
