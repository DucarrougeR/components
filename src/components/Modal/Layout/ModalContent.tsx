import * as React from 'react'
import ReactResizeDetector from 'react-resize-detector'
import { palette, styled } from '../../../style'
import { Box, BoxProps } from '../../Box'

export interface ModalContentProps extends BoxProps<HTMLDivElement> {
  /**
   * Content that will be placed inside the DialogHeader
   * @required
   */
  children: React.ReactNode
  /**
   * Style the scrollable space within the ModalContent.
   * Often p="none" is applied if componennts within the the ModalContent need to the
   * touch the container edges.
   */
  innerProps?: BoxProps<HTMLDivElement>
}

interface ContentState {
  overflow: boolean
}

interface InternalContentProps extends ModalContentProps {
  renderedHeight: string
}

class Internal extends React.Component<InternalContentProps, ContentState> {
  private ref: React.RefObject<HTMLDivElement>

  constructor(props: InternalContentProps) {
    super(props)
    this.state = { overflow: false }
    this.ref = React.createRef()
  }

  public hasOverflow(e: HTMLDivElement) {
    return e.offsetHeight < e.scrollHeight
  }

  public componentDidUpdate(prevProps: InternalContentProps) {
    if (prevProps.renderedHeight !== this.props.renderedHeight) {
      this.ref.current &&
        this.setState({ overflow: this.hasOverflow(this.ref.current) })
    }
  }

  public render() {
    const { children, renderedHeight, innerProps, ...props } = this.props

    return (
      <ContentContainer
        overflow="auto"
        className={`${props.className && props.className} ${this.state
          .overflow && 'overflow'}`}
        innerRef={this.ref}
        flex="8"
        {...props}
      >
        <Box p="large" {...innerProps}>
          {children}
        </Box>
      </ContentContainer>
    )
  }
}

export const ModalContent: React.SFC<ModalContentProps> = props => {
  return (
    <ReactResizeDetector handleHeight>
      {(height: string) => <Internal renderedHeight={height} {...props} />}
    </ReactResizeDetector>
  )
}

const ContentContainer = styled(Box)`
  &.overflow {
    box-shadow: inset 0 -16px 16px -16px ${palette.charcoal200},
      inset 0 16px 16px -16px ${palette.charcoal200};
  }
`
