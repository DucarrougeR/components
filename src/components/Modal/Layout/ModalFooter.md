`ModalFooter` is attached to the bottom of modal. Generally, the terminal action for a modal as well as an option to abandon the process done via the modal are available within this component. This component is generally used in combination with the `ModalContext.Provider` to get access to the `ModalContext.closeModal()` method.

ModalFooter supports all [`Box`](/#!/Box) properties.

```js
<ModalFooter>
  <Button mx="medium" variant="transparent">
    Cancel
  </Button>
  <Button>Do stuff...</Button>
</ModalFooter>
```
