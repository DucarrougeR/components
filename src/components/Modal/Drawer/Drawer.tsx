import * as React from 'react'
import { Modal, ModalProps } from '../Modal'
import { DrawerSurface } from './DrawerSurface'

export interface DrawerProps extends ModalProps {
  /**
   * Content that will be placed inside the Dialog
   * @required
   */
  children: React.ReactNode
}

export const Drawer: React.SFC<DrawerProps> = ({
  width,
  children,
  surfaceStyles,
  ...props
}) => {
  const surface = (animationState: string) => {
    return (
      <DrawerSurface
        style={surfaceStyles}
        className={animationState}
        width={width}
      >
        {children}
      </DrawerSurface>
    )
  }
  return <Modal {...props} render={surface} />
}
