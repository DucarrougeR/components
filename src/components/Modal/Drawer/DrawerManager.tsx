import * as React from 'react'
import { Drawer } from '.'
import { ManagedModalProps } from '../Modal'
import { ModalManager, ModalManagerProps } from '../ModalManager'

export class DrawerManager extends ModalManager<ModalManagerProps> {
  protected renderModal(content: React.ReactNode, props: ManagedModalProps) {
    return (
      <Drawer isOpen={this.state.isOpen} onClose={this.close} {...props}>
        {content}
      </Drawer>
    )
  }
}
