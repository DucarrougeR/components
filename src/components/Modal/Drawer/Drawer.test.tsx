import 'jest-styled-components'
import * as React from 'react'
import {
  assertSnapshot,
  assertSnapshotShallow,
} from '../../../../test/utils/snapshot'
import { Button } from '../../Button'
import { Heading } from '../../Heading'
import { ModalContent, ModalFooter, ModalHeader } from '../Layout'
import { Drawer } from './Drawer'
import { DrawerManager } from './DrawerManager'

const content = (
  <>
    words and stuff <button>boo!</button>
  </>
)

test('Drawer Hidden', () => {
  assertSnapshot(
    <DrawerManager content={content}>
      {onClick => <Button onClick={onClick}>ðŸ¥‘</Button>}
    </DrawerManager>
  )
})

test('Drawer Shown', () => {
  assertSnapshotShallow(<Drawer isOpen>{content}</Drawer>)
})

test('Drawer, backdrop customized', () => {
  assertSnapshotShallow(
    <Drawer isOpen backdropStyles={{ background: 'purple', opacity: 1 }}>
      {content}
    </Drawer>
  )
})

test('Selection Drawer, Shown', () => {
  assertSnapshotShallow(
    <Drawer isOpen>
      <ModalHeader>
        <Heading>Pick stuff</Heading>
      </ModalHeader>
      <ModalContent>Stuff and things...</ModalContent>
      <ModalFooter>
        <Button variant="transparent" mr="small">
          Cancel
        </Button>
        <Button>Pick some stuff</Button>
      </ModalFooter>
    </Drawer>
  )
})
