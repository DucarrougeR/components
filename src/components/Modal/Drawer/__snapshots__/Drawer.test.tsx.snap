// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Hidden 1`] = `
.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1rem;
  line-height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c0 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  border-style: solid;
  border-width: 0.0625rem;
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

<button
  className="c0 c1"
  onClick={[Function]}
>
  ðŸ¥‘
</button>
`;

exports[`Drawer Shown 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  background-color: #DEE1E5;
  height: 100%;
  opacity: 0.6;
  position: fixed;
  width: 100%;
}

.c5 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  box-shadow: 0 3px 18px rgba(0,0,0,0.12),0 1px 4px rgba(0,0,0,0.04);
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  max-width: 100%;
  position: absolute;
  right: 0px;
  top: 0px;
  width: 30rem;
}

.c1 {
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
  cursor: pointer;
}

.c1.entering,
.c1.exiting {
  opacity: 0.01;
}

.c1.exited {
  opacity: 0.6;
}

.c4 {
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
  -webkit-transition: transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
  transition: transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
}

.c3.entering,
.c3.exiting {
  opacity: 0.01;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
}

.c3.exited {
  opacity: 1;
  -webkit-transform: translateX(0%);
  -ms-transform: translateX(0%);
  transform: translateX(0%);
}

<div
  className="c0"
>
  <div
    className="faux c1 c2"
    onClick={[Function]}
  />
  <div
    className="faux c3 c4 c5"
  >
    words and stuff
  </div>
</div>
`;

exports[`Drawer, backdrop customized 1`] = `
Array [
  .c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1rem;
  line-height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c0 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  border-style: solid;
  border-width: 0.0625rem;
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

<button
    className="c0 c1"
    onClick={[Function]}
  >
    ðŸ¥‘
  </button>,
  .c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  background-color: #DEE1E5;
  height: 100%;
  opacity: 0.6;
  position: fixed;
  width: 100%;
}

.c5 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  box-shadow: 0 3px 18px rgba(0,0,0,0.12),0 1px 4px rgba(0,0,0,0.04);
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  max-width: 100%;
  position: absolute;
  right: 0px;
  top: 0px;
  width: 30rem;
}

.c1 {
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
  cursor: pointer;
  background: purple;
  opacity: 1;
}

.c1.entering,
.c1.exiting {
  opacity: 0.01;
}

.c1.exited {
  opacity: 0.6;
}

.c4 {
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
  -webkit-transition: transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
  transition: transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
}

.c3.entering,
.c3.exiting {
  opacity: 0.01;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
}

.c3.exited {
  opacity: 1;
  -webkit-transform: translateX(0%);
  -ms-transform: translateX(0%);
  transform: translateX(0%);
}

<div
    className="c0"
  >
    <div
      className="faux c1 c2"
      onClick={[Function]}
    />
    <div
      className="faux c3 c4 c5"
    >
      words and stuff
    </div>
  </div>,
]
`;

exports[`Selection Drawer, Shown 1`] = `
Array [
  .c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1rem;
  line-height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c0 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  border-style: solid;
  border-width: 0.0625rem;
  background: #FFFFFF;
  border-color: #C1C6CC;
  color: #64518A;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #FFFFFF;
  border-color: #C1C6CC;
  color: #64518A;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #FFFFFF;
  border-color: #64518A;
  color: #2A1B60;
}

.c0:active,
.c0.active {
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

<button
    className="c0 c1"
    onClick={[Function]}
  >
    Pick Stuff
  </button>,
  .c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  background-color: #DEE1E5;
  height: 100%;
  opacity: 0.6;
  position: fixed;
  width: 100%;
}

.c5 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  box-shadow: 0 3px 18px rgba(0,0,0,0.12),0 1px 4px rgba(0,0,0,0.04);
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  max-width: 100%;
  position: absolute;
  right: 0px;
  top: 0px;
  width: 30rem;
}

.c6 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 1.25rem;
}

.c7 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin-right: 2rem;
}

.c9 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.75rem;
}

.c11 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1.375rem;
  line-height: 42px;
  margin-left: auto;
  padding-left: 1.25rem;
  padding-right: 1.25rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c12 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1em;
  width: 1em;
}

.c14 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-flex: 8;
  -ms-flex: 8;
  flex: 8;
  overflow: scroll;
  padding: 1.25rem;
}

.c15 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 1.25rem;
}

.c16 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1rem;
  line-height: 34px;
  margin-right: 0.75rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c18 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1rem;
  line-height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c1 {
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
  cursor: pointer;
}

.c1.entering,
.c1.exiting {
  opacity: 0.01;
}

.c1.exited {
  opacity: 0.6;
}

.c4 {
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
  -webkit-transition: transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
  transition: transform 300ms cubic-bezier(0.86,0,0.07,1), opacity 300ms cubic-bezier(0.86,0,0.07,1);
}

.c3.entering,
.c3.exiting {
  opacity: 0.01;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
}

.c3.exited {
  opacity: 1;
  -webkit-transform: translateX(0%);
  -ms-transform: translateX(0%);
  transform: translateX(0%);
}

.c17 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  border-style: solid;
  border-width: 0.0625rem;
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

.c17:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c17[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c17[disabled]:hover,
.c17[disabled]:active,
.c17[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

.c17:hover,
.c17:focus,
.c17.hover {
  background: #473764;
  border-color: #473764;
}

.c17:active,
.c17.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c10 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  border-style: solid;
  border-width: 0.0625rem;
  background: transparent;
  border-color: transparent;
  color: #64518A;
}

.c10:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c10[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c10[disabled]:hover,
.c10[disabled]:active,
.c10[disabled]:focus {
  background-color: transparent;
  border-color: transparent;
  color: #64518A;
}

.c10:hover,
.c10:focus,
.c10.hover {
  background: #F3F2FF;
  border-color: #F3F2FF;
  color: #6C43E0;
}

.c10:active,
.c10.active {
  background: #E8E5FF;
  border-color: #E8E5FF;
  color: #6C43E0;
}

.c8 {
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-transform: none;
}

.c13 {
  border-top: 1px solid transparent;
}

.c13.overflow {
  border-top-color: #DEE1E5;
  box-shadow: inset 0 -16px 16px -16px #DEE1E5;
}

<div
    className="c0"
  >
    <div
      className="faux c1 c2"
      onClick={[Function]}
    />
    <div
      className="faux c3 c4 c5"
    >
      <header
        className="c6"
      >
        <div
          className="c7"
        >
          <h2
            className="c8 c9"
          >
            Pick stuff
          </h2>
        </div>
        <button
          className="c10 c11"
          onClick={[Function]}
          style={
            Object {
              "lineHeight": 1,
              "padding": 0,
            }
          }
        >
          <div
            className="c12"
          >
            <svg
              fill="currentColor"
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
                fill="currentColor"
              />
            </svg>
          </div>
        </button>
      </header>
      <div
        style={
          Object {
            "display": "none",
            "height": 0,
            "position": "absolute",
            "visibility": "hidden",
            "width": 0,
          }
        }
      />
      <div
        className="undefined false c13 c14"
        overflow="scroll"
      >
        Stuff and things...
      </div>
      <footer
        className="c15"
      >
        <button
          className="c10 c16"
        >
          Cancel
        </button>
        <button
          className="c17 c18"
        >
          Pick some stuff
        </button>
      </footer>
    </div>
  </div>,
]
`;
